docs/src/main/tut/TypedDatasetVsSparkDataset.md

```diff
@@ -131,8 +131,42 @@ And the compiler is our friend.
 fds.filter( fds('i) === 10 ).select( fds('x) )
 ```
 
+## Differences in Encoders
+
+Encoders in Spark's `Datasets` are partially type-safe. If you try to create a `Dataset` using  a type that is not 
+ a Scala `Product` then you get a compilation error:
```

Comment:
@OlivierBlanvillain Are you talking about classes that extends `Product` explicitly ? If it's the case, spark does handle them.